lte/gateway/c/core/oai/include/state_manager.h

```diff
@@ -165,27 +175,86 @@ class StateManager {
         is_initialized,
         "StateManager init() function should be called to initialize state");
 
+    auto start = std::chrono::high_resolution_clock::now();
     std::string proto_str;
     ProtoUe ue_proto = ProtoUe();
     StateConverter::ue_to_proto(ue_context, &ue_proto);
     redis_client->serialize(ue_proto, proto_str);
     std::size_t new_hash = std::hash<std::string>{}(proto_str);
-
+    auto stop            = std::chrono::high_resolution_clock::now();
+    std::cout << "TIME PROTOBUF CONVERSION : "
+              << (std::chrono::duration_cast<std::chrono::nanoseconds>(
+                      stop - start))
+                     .count()
+              << std::endl;
     if (new_hash != this->ue_state_hash[imsi_str]) {
       std::string key = IMSI_PREFIX + imsi_str + ":" + task_name;
+
+      start = std::chrono::high_resolution_clock::now();
       if (redis_client->write_proto_str(
               key, proto_str, ue_state_version[imsi_str]) != RETURNok) {
         OAILOG_ERROR(
             log_task, "Failed to write UE state to db for IMSI %s",
             imsi_str.c_str());
         return;
       }
+      stop = std::chrono::high_resolution_clock::now();
+      std::cout << "TIME PROTOBUF REDIS SERIALIZATION : "
+                << std::chrono::duration_cast<std::chrono::nanoseconds>(
+                       stop - start)
+                       .count()
+                << std::endl;
+
+      this->ue_state_version[imsi_str]++;
+      this->ue_state_hash[imsi_str] = new_hash;
+      OAILOG_DEBUG(
+          log_task, "Finished writing UE state for IMSI %s", imsi_str.c_str());
+    }
+  }
+  /**
+   * Writes task state to db if persist_state is enabled
+   */
+  virtual void write_ue_state_to_db(
+      uint8_t* buf, int size, const std::string& imsi_str) {
+    AssertFatal(
+        is_initialized,
+        "StateManager init() function should be called to initialize state");
+    // lack of C++17 (hashing a stringview)
+    auto start = std::chrono::high_resolution_clock::now();
```

Comment:
are you planning to add the benchmark cmake test flag for these functions? on the previous PR I think we agreed to add a flag to not compile the implementation code for benchmarking tests

Reply:
So we have to remove benchmark of redis serialization.

Reply:
The purpose is not to remove benchmark of redis serialization, but to not introduce changes to the production code that could affect performance (We do not want to benchmark and run this analysis everytime that we are running the MME service)

Follow the example on https://github.com/magma/magma/pull/8487/files#diff-41d3ad27a8ccc4fa615212321c1f0d4d49af2a4920598f9c1283ea1856c8820b, I'd suggest adding a new flag `OAI_BENCHMARK_FLAGS = -DMME_BENCHMARK=True` and add a new `make benchmark_fb_pb` command to use that flag, this way you can enclose all the benchmark code with `#if MME_BENCHMARK`

Reply:
done